code/modules/paperwork/photocopier.dm

```diff
@@ -276,47 +140,320 @@
+				break
+	else if(istype(C, /obj/item/paper_bundle))
+		playsound(loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
+		var/obj/item/paper_bundle/I = copyitem
+		for(var/i = copies; i > 0; i--)
+			if(toner < (I.amount + 1))
+				visible_message("<span class='notice'>A yellow light on [src] flashes, indicating there's not enough toner to finish the operation.</span>")
+				return // It is better to prevent partial bundle than to produce broken paper bundle
+			var/obj/item/paper_bundle/B = bundlecopy(C)
+			if(!B) //B returned null because it was partial
+				break
+			sleep(PHOTOCOPIER_DELAY*B.amount)
+	else if(M && M.loc == src.loc)
+		playsound(loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
+		for(var/i = copies; i > 0; i--)
+			if(toner > 4)
+				copyass()
+				sleep(PHOTOCOPIER_DELAY)
+			else
+				visible_message("<span class='notice'>A yellow light on [src] flashes, indicating there's not enough toner to finish the operation.</span>")
+				break
+	else
+		to_chat(usr, "<span class='warning'>\The [copyitem] can't be copied by \the [src].</span>")
+	copying = FALSE
+	SStgui.update_uis(src)
+
+/obj/machinery/photocopier/proc/scan_document() //scan a document into a file
+	if(!cancopy())
+		return
+	if(saved_documents.len >= max_saved_documents)
+		to_chat(usr, "<span class='warning'>\The [copyitem] can't be scanned because the max file limit has been reached. Please delete a file to make room.</span>")
+		return
+	copying = TRUE
+	var/obj/item/O
+	if(istype(copyitem, /obj/item/paper))
+		O = papercopy(copyitem, scanning = TRUE)
+	else if(istype(copyitem, /obj/item/photo))
+		O = photocopy(copyitem, scanning = TRUE)
+	else if(istype(copyitem, /obj/item/paper_bundle))
+		O = bundlecopy(copyitem, scanning = TRUE)
+	else if(M && M.loc == src.loc)
+		O = copyass(scanning = TRUE)
+	else
+		to_chat(usr, "<span class='warning'>\The [copyitem] can't be scanned by \the [src].</span>")
+		copying = FALSE
+		return
+	use_power(active_power_usage)
+	sleep(PHOTOCOPIER_DELAY)
+	saved_documents.Add(O)
+	copying = FALSE
+	playsound(loc, 'sound/machines/ping.ogg', 50, 0)
+	atom_say("Document succesfully scanned!")
+
+/obj/machinery/photocopier/proc/delete_file(uid)
+	var/document = locateUID(uid)
+	if(document in saved_documents)
+		saved_documents.Remove(document)
+		qdel(document)
+		SStgui.update_uis(src)
+
+/obj/machinery/photocopier/proc/file_copy(uid)
+	var/document = locateUID(uid)
+	if(document in saved_documents)
+		copy(document, scancopy=TRUE)
+
+
+/obj/machinery/photocopier/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "Photocopier", name, 402, 368, master_ui, state)
+		ui.set_autoupdate(TRUE)
```

Comment:
Did you want false here? Because it seems you're going the manual route of updating with `SStgui.update_uis(src)`
```suggestion
		ui.set_autoupdate(FALSE)
```

Reply:
Ah, wasn't 100% what this did. I'm going to go with autoupdating for this.

Reply:
This is on by default, so this line can be removed.